test/src/test/java/aima/test/unit/search/basic/BasicPriorityFrontierQueueTest.java

```diff
@@ -0,0 +1,131 @@
+package aima.test.unit.search.basic;
+
+import aima.core.search.basic.support.BasicFrontierQueue;
+import aima.core.search.basic.support.BasicNodeFactory;
+import aima.core.search.basic.support.BasicPriorityFrontierQueue;
+import aima.core.search.api.Node;
+import aima.core.search.api.NodeFactory;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.PriorityQueue;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BasicPriorityFrontierQueueTest<A, S> {
+	
+	private static NodeFactory<Integer, String> nodeFactory;
+	private static int NUMBER_OF_NODES;
+	private static int NUMBER_OF_STATES;
+	private static Node[] testNodes;
+	private static Node[] fastPQResult;
+	private static Node[] normalPQResult;
+	private static Comparator<Node<Integer, String>> comparator;
+	
+	@BeforeClass
+	public static void prepare() {
+		
+		
+		nodeFactory = new BasicNodeFactory<Integer, String>();
+		
+		//Custom comparator to avoid duplicates in a TreeSet
+		//Checks on pathCost first, followed by state (only for a clearer output), and finally followed by object equality
+		
+		comparator = (Node<Integer, String> node1, Node<Integer, String> node2)-> 
+					 ((node1.pathCost() == node2.pathCost()) ? ((node1.state().equals(node2.state())) ? (node1.equals(node2) ? 0 : 1) : (node1.state().compareTo(node2.state()))) 
+					 : (int)Math.signum((node1.pathCost() - node2.pathCost())));
+		

```

Comment:
A possibly more compact/readable version of this would be:

comparator = Comparator			
		.comparing(Node<Integer, String>::pathCost)
		.thenComparing(Comparator.comparing(Node<Integer, String>::state))
		.thenComparing((n1, n2) -> Objects.equals(n1, n2) ? 0 : 1);